### ISSUE TITLE
1.0.0-beta.10: Failed to construct 'URL': Invalid URL

### ISSUE THREAD

**Issue by DaveXCS** [auth0/angular2-jwt#477]
with beta 9 everything was working fine but with beta10 i get now:
Failed to construct 'URL': Invalid URL
url example:
`/api/v1/system/mode` (same server)
```
JwtModule.forRoot({
      config: {
        tokenGetter: tokenGetter,
        whitelistedDomains: ['localhost']
      }
    }),
```

```
export function tokenGetter() {
  return localStorage.getItem(Configuration.AUTH_TOKEN_ID);
}
```

This issue is caused by the changes on src/jwt.interceptor.ts in pull request #438 
`const requestUrl = new URL(request.url);`
will not accept relative urls. It has to be `new URL(relative, base) `or `new URL(full-url)`

**Comment by laran** [comment]
+1

**Comment by jameskentTX** [comment]
I have the same error.

**Comment by miguelcaravantes** [comment]
same issue

**Comment by cmcclellen** [comment]
Any word on this?  Did you find any work-arounds?

**Comment by fititrick** [comment]
same issue

**Comment by DaveXCS** [comment]
The current workaround is to use full url or use beta9. 
The author of the pull request has the opinion that a crashing or non working application is better than an unsafe application.
I would personally prefer to make it safe and working but it seems they have different understanding about that :-)

**Comment by Mike1707** [comment]
@DaveXCS Unfortunatly, I don't get it working. Maybe it's an issue because I'm using the Ionic Framework (3.9.2)? At the moment, I'm doing this:

`whitelistedDomains: [new URL('http://localhost:1337'), new URL('https://api.myurl.com')]`

But the same Error: `TypeError: Failed to construct 'URL': Invalid URL`

With beta.9 it was:

`whitelistedDomains: ['localhost:1337', 'api.myurl.com']`

**Comment by DaveXCS** [comment]
The problem is not the url of the whitlisted domains. The problem is if you send any request with the angular http service e.g.:
`this.http.get('/api/values', httpOptions)`
will not work but
`this.http.get('http://localhost:3030/app/api/values', httpOptions)`
will work.
Beta9 had a try catch around `const requestUrl = new URL(request.url);`
So in the case of a relative url the error 'Invalid URL' was catched with an empty catch block.
In Beta10 they removed the try/catch and now relative url requests are no longer working (which is the default on most of the application).

**Comment by Mike1707** [comment]
That's very interesting since I'm totally sure I just use full URLs in all of my requests.

**Comment by mjb1632** [comment]
At the point the interceptors are called, how do you ever know the full path?
I want an application to be able to work on dev, test, prod servers and domain names might change so I never want any specific domain hard coded.  I suppose I need to make a non-intercepted request first to get the baseUrl of the deployment and then use that?

**Comment by DaveXCS** [comment]
you could do here sth like -> if request.url starts with '/' (or regex expression) and whitlisted domains contain 'localhost' -> return true so you don't need the full url.

btw you can write your own interceptor instead of the angular-jwt interceptor sth like that:

```
@Injectable()
export class TokenInterceptor implements HttpInterceptor {
  constructor(public auth: AuthService) {}
  intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
    
    request = request.clone({
      setHeaders: {
        Authorization: `Bearer ${this.auth.getToken()}`
      }
    });
    return next.handle(request);
  }
}
```

**Comment by brian428** [comment]
There are also cases where it's impossible to know the full URL in advance (the final prod host name is unknown at development time). I think the library **has** to allow for relative URLs.

**Comment by Sambaja** [comment]
Could we replace the creation of the url by this :

```js
var requestUrl = new URL(request.url, /^https?:\/\//i.test(request.url) ? undefined : window.location.href)
```

**Comment by anibalsanchez** [comment]
Same issue here. 

Sample URL: "./assets/i18n/en.json"

**Comment by dushkostanoeski** [comment]
Why hasn't this been addressed before the final release?

**Comment by noumaans** [comment]
@Sambaja we are using this library under {N}ativeScript on iOS / Android. I'd prefer the fix in #489 if possible.

**Comment by RedAB** [comment]
Hi everybody,

I was dealing with exactly the same issue after updating to 1.0.0, even if I was already using absolute URLs in all of my requests. After investigating, I realize that the problem came from my custom translation package TranslateModule factory.

```
export function TranslateFactory(http: HttpClient): TranslateHttpLoader {

    return new TranslateHttpLoader(http, './assets/i18n/', '.json');
}
```

And this is how I fixed it.

export function TranslateFactory(http: HttpClient): TranslateHttpLoader {

    // Absolute URL must be specified for authentication whitelisted domains
    return new TranslateHttpLoader(http, location.origin + '/assets/i18n/', '.json');
}

Hope this will be helpful!

**Comment by jualoppaz** [comment]
That's right. It's a very good solution by the moment. Thank you so much!!

**Comment by elfrasco** [comment]
Same issue :(

**Comment by ydemir** [comment]
Same issue :(

Meanwhile I had to do a downgrade to 1.0.0-beta.9 version.

**Comment by anibalsanchez** [comment]
The workaround with `location.origin` works perfectly fine... Still, it should only whitelist URIs with a domain, and not throw errors with any other URIs.

**Comment by aaronte** [comment]
Same problem with release 1.0.0 :(

**Comment by liofal** [comment]
Same problem with release 1.0.0

**Comment by valera33** [comment]
Can confirm, same problem with release 1.0.0 :(

**Comment by kadavre** [comment]
Same issue here, reverting to beta 9 :/

**Comment by thir13en** [comment]
Same issue, fallback to beta.9 did work as a charm

**Comment by aamir1995** [comment]
This should be fixed with the release of `1.1.0`.

**Comment by anibalsanchez** [comment]
Confirmed. Solved with 1.1.0.

**Comment by thir13en** [comment]
Thanks! Will upvote as soon as I sign in GitHub!

Il 16 mar 2018 5:59 PM, "Anibal Sanchez" <notifications@github.com> ha
scritto:
